warehouse/utils/utils.go

```diff
@@ -1075,8 +1075,16 @@ func StagingTablePrefix(provider string) string {
 }
 
 func StagingTableName(provider, tableName string, tableNameLimit int) string {
-	randomNess := strings.ReplaceAll(uuid.Must(uuid.NewV4()).String(), "-", "")
+	randomNess := RandHex()
 	prefix := StagingTablePrefix(provider)
 	stagingTableName := fmt.Sprintf(`%s%s_%s`, prefix, tableName, randomNess)
 	return misc.TruncateStr(stagingTableName, tableNameLimit)
 }
+
+// RandHex returns a random hex string of length 32
+func RandHex() string {
```

Comment:
Is there any improvement in this function in terms of speed or uniqueness over the other function ? Just curious to understand the motivation behind it.

Reply:
Yes. Actually, UUID.string() is processing for 36 bytes. We are ignoring the `-` present by replacing it using `strings.ReplaceAll(uuid.Must(uuid.NewV4()).String(), "-", "")`. We can just probably call `hex.Encode` which can just populate the 32 bytes.

```
const Size = 16
type UUID [Size]byte
```

```
func (u UUID) String() string {
	buf := make([]byte, 36)

	hex.Encode(buf[0:8], u[0:4])
	buf[8] = '-'
	hex.Encode(buf[9:13], u[4:6])
	buf[13] = '-'
	hex.Encode(buf[14:18], u[6:8])
	buf[18] = '-'
	hex.Encode(buf[19:23], u[8:10])
	buf[23] = '-'
	hex.Encode(buf[24:], u[10:])

	return string(buf)
}
```